src/clientrpc.cpp

```diff
@@ -0,0 +1,166 @@
+        QJsonObject result{ { "connected", pClient->IsConnected() } };
+        response["result"] = result;
+    } );
+
+    pRpcServer->HandleMethod ( "jamulusclient/getChannelInfo", [=] ( const QJsonObject& params, QJsonObject& response ) {
+        QJsonObject result{ { "name", pClient->ChannelInfo.strName }, { "skillLevel", FormatSkillLevel ( pClient->ChannelInfo.eSkillLevel ) } };
+        response["result"] = result;
+    } );
+
+    pRpcServer->HandleMethod ( "jamulusclient/getClientList", [=] ( const QJsonObject& params, QJsonObject& response ) {
+        QJsonObject result{ { "clients", arrStoredChanInfo } };
+        response["result"] = result;
+    } );
+
+    pRpcServer->HandleMethod ( "jamulusclient/setName", [=] ( const QJsonObject& params, QJsonObject& response ) {
+        auto jsonName = params["name"];
+        if ( !jsonName.isString() )
+        {
+            response["error"] = CreateJsonRpcError ( -32602, "Invalid params" );
+            return;
+        }
+
+        pClient->ChannelInfo.strName = jsonName.toString().left ( MAX_LEN_FADER_TAG );
+        pClient->SetRemoteInfo();
+
+        response["result"] = "ok";
+    } );
+
+    pRpcServer->HandleMethod ( "jamulusclient/setSkillLevel", [=] ( const QJsonObject& params, QJsonObject& response ) {
+        auto jsonSkillLevel = params["skillLevel"];
+        if ( jsonSkillLevel.isNull() )
+        {
+            pClient->ChannelInfo.eSkillLevel = SL_NOT_SET;
+            pClient->SetRemoteInfo();
+            return;
+        }
+
+        if ( !jsonSkillLevel.isString() )
+        {
+            response["error"] = CreateJsonRpcError ( -32602, "Invalid params" );
+            return;
+        }
+
+        auto strSkillLevel = jsonSkillLevel.toString();
+        if ( strSkillLevel == "beginner" )
+        {
+            pClient->ChannelInfo.eSkillLevel = SL_BEGINNER;
+        }
+        else if ( strSkillLevel == "intermediate" )
+        {
+            pClient->ChannelInfo.eSkillLevel = SL_INTERMEDIATE;
+        }
+        else if ( strSkillLevel == "expert" )
+        {
+            pClient->ChannelInfo.eSkillLevel = SL_PROFESSIONAL;
+        }
+        else
+        {
+            response["error"] = CreateJsonRpcError ( -32602, "Invalid params" );
+            return;
+        }
+
+        pClient->SetRemoteInfo();
+        response["result"] = "ok";
+    } );
+
+    pRpcServer->HandleMethod ( "jamulusclient/sendChatText", [=] ( const QJsonObject& params, QJsonObject& response ) {
+        auto jsonMessage = params["message"];
+        if ( !jsonMessage.isString() )
+        {
+            response["error"] = CreateJsonRpcError ( -32602, "Invalid params" );
+            return;
+        }
+
+        pClient->CreateChatTextMes ( jsonMessage.toString() );
+
+        response["result"] = "ok";
+    } );
+}
+
+CClientRpc::~CClientRpc() {}
+
+static QJsonValue FormatSkillLevel ( ESkillLevel eSkillLevel )
```

Comment:
I feel this should live very close to where `ESkillLevel` is defined.

Reply:
Done — moved to `CClientRpc::SerializeSkillLevel`

8f7c0960ccf091186e187c638b37ad4187427fbf